app/components/views/Offset/index.tsx

```diff
@@ -170,6 +172,11 @@ export const Offset = (props: Props) => {
 
   // effects
   useEffect(() => {
+    // round the value for the edge case where a user inputs a fractional value
+    // when fiat isn't selected to pay. Example (100.5 -> 101)
+    if (paymentMethod === "fiat" && !Number.isInteger(parseFloat(quantity))) {
+      setQuantity(Math.round(Number(quantity)).toString());
+    }
```

Comment:
we can be more concise since setting state is cheap
```suggestion
    if (paymentMethod === "fiat") {
      setQuantity(Math.ceil(quantity).toString());
    }
```

also I'd go with `ceil` because, with carbon offsetting, you are generally wanting to cover **all** of a given emissions quantity.

edit: I think doing it this way will actually solve the query params issue. If so, please document that in a comment here.

Reply:
Thanks for all the suggestions, updated the PR and addressed all comments. This check covers the case for when we switch payment provider. However, it didn't cover the case for query params as there is another update that occurs after this update in an effect that handles query params specifically, so I've added a check around this also.